tests/functional/object/test_object_browser_client_provider_ui.py

```
@@ -500,3 +512,359 @@ def test_object_browser_list_objects_with_folders(self, project_factory):
+
+                # Extract S3 credentials
+                s3_creds = get_s3_credentials_from_obc(obc_name, namespace)
+                bucket_name = s3_creds["bucket_name"]
+                logger.info("Bucket: %s", bucket_name)
+
+                # Create S3 client
+                s3_client = boto3.client(
+                    "s3",
+                    aws_access_key_id=s3_creds["access_key_id"],
+                    aws_secret_access_key=s3_creds["secret_access_key"],
+                    endpoint_url="https://%s" % s3_creds["endpoint"],
```

Ruff (0.15.21)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)